internal/convert/echo.go

```diff
@@ -33,6 +33,19 @@ func HalToEcho(h *config.Hal) *config.Echo {
 		Gcb:          h.GetCi().GetGcb(),
 		Stats:        getEchoStats(h),
 		Scheduler:    getEchoScheduler(h),
+		Services:     getEchoServices(h),
+	}
+}
+
+func getEchoServices(h *config.Hal) *config.Echo_Services {
+	if !h.GetManagedDelivery().GetEnabled().GetValue() {
+		return nil
+	}
+
+	return &config.Echo_Services{
+		Keel: &config.ServiceSettings{
+			Enabled: h.GetManagedDelivery().GetEnabled(),
```

Comment:
Rather than including this block in the several places that need it, would it be possible to have users set `services.keel.enabled` in their service discovery file as documented for other services [here](https://github.com/spinnaker/kustomization-base#enable-optional-services).

When we looked at the first two optional services (`fiat` and `kayenta`) we leaned towards not having `kleat` do too much overriding of the service discovery file (the `spinnaker.yml`) figuring that that was more of a deployment concern.  (If you are enabling disabling services you'll also need to actually deploy/undeploy those.)

I think there are probably some improvements we can make to streamline the workflow, but as a first pass I'd lean towards not having `kleat` write out the `services.*` block to the service config.  Does that sound reasonable?

Reply:
+1, and let's also consider adding documentation on enabling `keel` to the [section](https://github.com/spinnaker/kustomization-base#enable-optional-services) that documents enabling `fiat` and `kayenta`.

Reply:
on the above, service setting for Kayenta seems to be enabled for orca and gate if `halconfig` has `h.GetCanary().GetEnabled()` set to `true` (see [Here](https://github.com/spinnaker/kleat/blob/master/internal/convert/orca.go#L50-L59)). And frankly speaking, I like it better  than to have to do extra tweaking and setting of the configuration files in an override.

is there an ordering concern here, that makes the override a preferred choice? if not, then for `Keel` also, if we enable it similar to how we do for Kayenta and make it depend on the value of `h.GetManagedDelivery().GetEnabled()`, it will be less tweaking of the configuration files and easier for the end users. wdyat?

Reply:
Thanks for pointing that out @nimakaviani, I had completely forgotten where we landed on this, but based on this [PR](https://github.com/spinnaker/kleat/pull/67) from May (which feels like it was from 5 years ago instead of 5 months ago!), I think you're right -- looks like we decided to shift the `enabled` flag into any service that cares whether an optional service is enabled based on the halconfig input, but otherwise all other service discovery information should be manually overridden. Does that sound right to you, @ezimanyi?

Reply:
ok great, if that is the case then I will update this PR to follow the same conventions. thanks!

Reply:
@ezimanyi, ping on the above and also on https://github.com/spinnaker/kustomization-base/pull/32.

Reply:
great, I think this PR is good to go then. I will fix spinnaker/kustomization-base#32 as well.

Reply:
Thanks! I'm happy to merge this PR now, but we should probably hold off on merging the corresponding `kustomization-base` PR until the `keel` Docker image lives in a Spinnaker-owned repository.

Reply:
@maggieneterval I think this is now unblocked because https://github.com/spinnaker/kustomization-base/pull/32 is submitted.

Since there has been some time since this thread was last ping'd. Are there any more things @nimakaviani would have to do before this could be merged? Thanks so much!

Reply:
Thanks, @achew22! Since @ncknt and @german-muzquiz are the new code owners, so I will defer to them for final approval and merge.